lib/Sema/ConstraintGraph.cpp

```diff
@@ -674,7 +674,7 @@ namespace {
             if (validComponents.count(inAdj) == 0)
               continue;
 
-            component.dependsOn.push_back(getComponent(inAdj).solutionIndex);
+            component.recordDependency(new Component(getComponent(inAdj)));
```

Comment:
One more thing - I think what you want here is a pointer to already existing component not a new component allocated in a global heap

Reply:
Hm, looks like what we actually return here is `flatComponents` so it would be incorrect to get pointers to `components` array :/ Looks like we'd have to use `SmallVector<unsigned, 2>` for dependencies after all... API could stay the same with a slight change that `recordDependency` accepts `const Component &` and `getDepencencies()` returns `ArrayRef<unsigned>`